puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katelloInstallerRelease.groovy

```diff
@@ -0,0 +1,45 @@
+pipeline {
+    agent { label 'admin' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Setup Environment') {
+            steps {
+                dir('katello-installer') {
+                    configureRVM()
+                    git url: 'https://github.com/katello/katello-installer', branch 'master'
+                    withRVM(['bundle install --without development --retry 5'])
+                    sh 'rm Puppetfile.lock'
+                }
+            }
+        }
+
+        stage('Build Tarball') {
+            steps {
+                dir('katello-installer') {
+                    withRVM(['bundle exec rake pkg:generate_source'])
+                    archiveArtifacts artifacts: 'pkg/*'
+                }
+            }
+        }
+
+        stage('Trigger RPM Build') {
+            steps {
+                build job: 'packaging_build_rpm', propagate: true, parameters: [
+                    string(name: 'project', value: 'packages/katello/katello-installer'),
+                    booleanParam(name: 'gitrelease', value: false),
+                    booleanParam(name: 'scratch', value: false),
+                    string(name: 'releaser', value: 'koji-katello-jenkins'),
+                    string(name: 'nightly_jenkins_job', value: 'release_build_katello_installer'),
+                    string(name: 'nightly_jenkins_job_id', value: 'lastSuccessfulBuild')
```

Comment:
Nope. According to `build_rpm_katello-installer_nightly`, it's just a string, not a variable.

Reply:
Correct the current job does this today because there are two separate jobs. If you self-reference the `lastSuccessfulBuild` it will go find the previous build given this current job is in progress already.

For example, say the last job was #4. This kicks off as #5, it creates an artifact, then it kicks off packaging_build_rpm and waits for it. Job #5 is in progress, so `lastSuccessfulBuild` at that time references #4 giving the wrong artifact.

Reply:
updated